packages/components/src/components/hds/icon/index.ts

```diff
@@ -104,12 +148,41 @@ export default class HdsIcon extends Component<HdsIconSignature> {
     if (this.predefinedColor) {
       classes.push(`hds-foreground-${this.predefinedColor}`);
     }
-
     // add an extra class to control the animation (depends on the icon)
     if (['loading', 'running'].includes(name)) {
       classes.push(`hds-icon--animation-${name}`);
     }
 
     return classes.join(' ');
   }
+
+  get viewBox(): string | undefined {
+    if (this.hdsCarbon.carbonModeEnabled) {
+      return this.carbonIcon?.attrs.viewBox;
+    } else {
+      return `0 0 ${this.size} ${this.size}`;
+    }
+  }
+
+  loadCarbonIcon = modifier((_element, [carbonModeEnabled]) => {
+    if (carbonModeEnabled) {
+      void this.loadCarbonIconTask.perform();
+    } else {
+      this.carbonIcon = null;
+    }
+  });
+
+  loadCarbonIconTask = task(async () => {
```

Comment:
[thought] what if, instead of loading the icons as unstructured JS objects and then build them at runtime as SVG, we leverage the existing icons pipeline to prepare them as Ember component (eg. as `.gts` template-only SVG files), and then import the component itself? this would also open the doors to do the same for the Flight icons, so you would not need two system of loading/showing icons, but only one.